### ISSUE TITLE
Failed to publish test results during release

### ISSUE THREAD

**Issue by andycodecomplove** [Microsoft/vsts-tasks#1898]
When I try to publish test results, that were generated in a previous task (and I know have been generated, because I can list the files in the folder to the console), I get the following error:

> Failed to publish test results: The underlying connection was closed: An unexpected error occurred on a send.

I'm using the the publish test results task that is provided (see attached image). Am I doing it right? Is there more information I can attach to help?

Cheers,
Andy

![image](https://cloud.githubusercontent.com/assets/17027173/16151748/996a1714-3496-11e6-9c2e-c8a7bfcd554b.png)

**Comment by andycodecomplove** [comment]
Update...

The release starts with an Azure Deployment task - a standard powershell script provided out of the box. 

If I move that to the end of the process, I don't get the error anymore.

I'm pretty sure I saw another issue with powershell scripts that was sensitive to the ordering of the scripts. I'll see if I can find it.

**Comment by andycodecomplove** [comment]
Here is the error that the Azure Deployment throws out:


2016-06-17T14:53:52.2596471Z Add-AzureRMAccount -ServicePrincipal -Tenant ******** -Credential System.Management.Automation.PSCredential
 
2016-06-17T14:53:52.5906475Z ##[error]Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException: Service returned error. Check InnerException for more details ---> System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a receive. ---> System.NullReferenceException: Object reference not set to an instance of an object.
 
2016-06-17T14:53:52.5916465Z ##[error]   at System.Management.Automation.ScriptBlock.InvokeWithPipe(Boolean useLocalScope, Dictionary`2 functionsToDefine, List`1 variablesToDefine, ErrorHandlingBehavior errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo invocationInfo, Object[] args)
 
2016-06-17T14:53:52.5916465Z ##[error]   at System.Management.Automation.ScriptBlock.InvokeAsDelegateHelper(Object dollarUnder, Object dollarThis, Object[] args)
 
2016-06-17T14:53:52.5926466Z ##[error]   at lambda_method(Closure , Object , X509Certificate , X509Chain , SslPolicyErrors )
 
2016-06-17T14:53:52.5926466Z ##[error]   at System.Net.ServerCertValidationCallback.Callback(Object state)
 
2016-06-17T14:53:52.5936510Z ##[error]   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
 
2016-06-17T14:53:52.5936510Z ##[error]   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
 
2016-06-17T14:53:52.5946514Z ##[error]   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
 
2016-06-17T14:53:52.5946514Z ##[error]   at System.Net.ServerCertValidationCallback.Invoke(Object request, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
 
2016-06-17T14:53:52.5956563Z ##[error]   at System.Net.Security.SecureChannel.VerifyRemoteCertificate(RemoteCertValidationCallback remoteCertValidationCallback)
 
2016-06-17T14:53:52.5966471Z ##[error]   at System.Net.Security.SslState.CompleteHandshake()
 
2016-06-17T14:53:52.5966471Z ##[error]   at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.5976482Z ##[error]   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.5986522Z ##[error]   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.5996499Z ##[error]   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.5996499Z ##[error]   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.6006473Z ##[error]   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.6107800Z ##[error]   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.6117819Z ##[error]   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
 
2016-06-17T14:53:52.6127801Z ##[error]   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
 
2016-06-17T14:53:52.6137784Z ##[error]   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
 
2016-06-17T14:53:52.6137784Z ##[error]   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
 
2016-06-17T14:53:52.6147806Z ##[error]   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
 
2016-06-17T14:53:52.6157792Z ##[error]   at System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)
 
2016-06-17T14:53:52.6157792Z ##[error]   at System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)
 
2016-06-17T14:53:52.6167809Z ##[error]   at System.Net.ConnectStream.WriteHeaders(Boolean async)
 
2016-06-17T14:53:52.6167809Z ##[error]   --- End of inner exception stack trace ---
 
2016-06-17T14:53:52.6177796Z ##[error]   at System.Net.HttpWebRequest.GetResponse()
 
2016-06-17T14:53:52.6177796Z ##[error]   at Microsoft.IdentityModel.Clients.ActiveDirectory.HttpWebRequestWrapper.<GetResponseSyncOrAsync>d__2.MoveNext()
 
2016-06-17T14:53:52.6187794Z ##[error]--- End of stack trace from previous location where exception was thrown ---
 
2016-06-17T14:53:52.6197803Z ##[error]   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 
2016-06-17T14:53:52.6197803Z ##[error]   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 
2016-06-17T14:53:52.6212249Z ##[error]   at Microsoft.IdentityModel.Clients.ActiveDirectory.HttpHelper.<SendPostRequestAndDeserializeJsonResponseAsync>d__0`1.MoveNext()
 
2016-06-17T14:53:52.6216634Z ##[error]   --- End of inner exception stack trace ---
 
2016-06-17T14:53:52.6226805Z ##[error]   at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.RunAsyncTask[T](Task`1 task)
 
2016-06-17T14:53:52.6226805Z ##[error]   at Microsoft.Azure.Commands.Common.Authentication.ServicePrincipalTokenProvider.GetAccessToken(AdalConfiguration config, ShowDialog promptBehavior, String userId, SecureString password, AccountType credentialType)
 
2016-06-17T14:53:52.6236786Z ##[error]   at Microsoft.Azure.Commands.Common.Authentication.Factories.AuthenticationFactory.Authenticate(AzureAccount account, AzureEnvironment environment, String tenant, SecureString password, ShowDialog promptBehavior, TokenCache tokenCache, Endpoint resourceId)
 
2016-06-17T14:53:52.6236786Z ##[error]   at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.AcquireAccessToken(AzureAccount account, AzureEnvironment environment, String tenantId, SecureString password, ShowDialog promptBehavior)
 
2016-06-17T14:53:52.6246810Z ##[error]   at Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient.Login(AzureAccount account, AzureEnvironment environment, String tenantId, String subscriptionId, String subscriptionName, SecureString password)
 
2016-06-17T14:53:52.6256798Z ##[error]   at Microsoft.Azure.Commands.Profile.AddAzureRMAccountCommand.ExecuteCmdlet()
 
2016-06-17T14:53:52.6256798Z ##[error]   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
 
2016-06-17T14:53:52.6266797Z ##[error]	ErrorCode: service_returned_error
 
2016-06-17T14:53:52.6276793Z ##[error]	StatusCode: 501
 
2016-06-17T14:53:52.6468828Z ##[error]System.Management.Automation.RuntimeException: There was an error with the service principal used for deployment.

**Comment by pavanadepu2** [comment]
@andycodecomplove , Could you provide the complete log file to analyze the issue further.

**Comment by pavanadepu2** [comment]
@andycodecomplove , am waiting on the complete logs. Can you provide it for analyzing the issue further.

**Comment by pavanadepu2** [comment]
@andycodecomplove, am closing the issue for now. Please reopen the issue when you are ready with the additional logs requested.
